serde-support/src/main/java/io/micronaut/serde/support/deserializers/StringListDeserializer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.serde.support.deserializers;
+
+import io.micronaut.core.type.Argument;
+import io.micronaut.serde.Decoder;
+import io.micronaut.serde.Deserializer;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * A specific collection of String.
+ *
+ * @author Denis Stepanov
+ */
+final class StringListDeserializer implements Deserializer<ArrayList<String>> {
```

Comment:
We should add more core list deserializers

Reply:
this can probably be a singleton with a static INSTANCE field